File: apps/Applications/ios/Local For Vocal/Local For Vocal/Sources/Views/Pages/PaymentView.swift
import SwiftUI

// MARK: - Color Hex Extension (Local Fallback)

public struct PaymentView: View {
    let totalAmount: Double
    let discount: Double
    let itemCount: Int
    let onBack: () -> Void
    let onPaymentSelect: (String) -> Void
    @Binding var isLoading: Bool

    @State private var expandedSection: String? = nil
    @State private var isTotalExpanded = false

    public var body: some View {
        VStack(spacing: 0) {
            // Header
            paymentHeader

            ScrollView(showsIndicators: false) {
                VStack(spacing: 0) {
                    // Total Amount Summary Card
                    totalAmountCard

                    // Discount Banner (if applicable)
                    if discount > 0 {
                        discountBanner
                    }

                    // Payment Options Title
                    HStack {
                        Text("Payment Options")
                            .font(.system(size: 14, weight: .semibold))
                            .foregroundColor(Color(hex: "#374151"))
                        Spacer()
                    }
                    .padding(.horizontal, 16)
                    .padding(.bottom, 12)

                    // Pay Online Card (Recommended)
                    payOnlineCard

                    // Saved Payment Options
                    PaymentAccordionView(
                        id: "saved",
                        icon: "clock",
                        title: "Saved Payment Options",
                        expandedSection: $expandedSection,
                        content: {
                            Text("No saved cards found.")
                                .font(.system(size: 13))
                                .foregroundColor(Color(hex: "#9CA3AF"))
                                .italic()
                        }
                    )

                    // Cash on Delivery
                    PaymentAccordionView(
                        id: "cod",
                        icon: "banknote",
                        title: "Cash on Delivery",
                        expandedSection: $expandedSection,
                        content: {
                            Button(action: {
                                onPaymentSelect("COD")
                            }) {
                                Text("Place Order")
                                    .font(.system(size: 14, weight: .bold))
                                    .foregroundColor(.white)
                                    .frame(maxWidth: .infinity)
                                    .frame(height: 44)
                                    .background(Color(hex: "#2563EB"))
                                    .cornerRadius(6)
                            }
                        }
                    )

                    Spacer().frame(height: 40)
                }
            }
        }
        .background(Color(hex: "#F3F4F6"))
        .navigationBarHidden(true)
    }

    // MARK: - Header
    private var paymentHeader: some View {
        HStack {
            HStack(spacing: 12) {
                Button(action: onBack) {
                    Image(systemName: "arrow.left")
                        .font(.system(size: 20, weight: .medium))
                        .foregroundColor(Color(hex: "#1F2937"))
                        .padding(4)
                }

                VStack(alignment: .leading, spacing: 2) {
                    Text("Step 3 of 3")
                        .font(.system(size: 12))
                        .foregroundColor(Color(hex: "#6B7280"))
                    Text("Payments")
                        .font(.system(size: 18, weight: .bold))
                        .foregroundColor(Color(hex: "#1F2937"))
                }
            }

            Spacer()

            // Secure Badge
            HStack(spacing: 4) {
                Image(systemName: "lock.fill")
                    .font(.system(size: 10))
                    .foregroundColor(Color(hex: "#4B5563"))
                Text("100% Secure")
                    .font(.system(size: 12, weight: .semibold))
                    .foregroundColor(Color(hex: "#4B5563"))
            }
            .padding(.horizontal, 8)
            .padding(.vertical, 4)
            .background(Color(hex: "#F3F4F6"))
            .cornerRadius(4)
        }
        .padding(.horizontal, 16)
        .padding(.vertical, 12)
        .background(Color.white)
        .overlay(
            Rectangle()
                .fill(Color(hex: "#E5E7EB"))
                .frame(height: 1),
            alignment: .bottom
        )
    }

    // MARK: - Total Amount Card
    private var totalAmountCard: some View {
        Button(action: {
            withAnimation(.easeInOut(duration: 0.25)) {
                isTotalExpanded.toggle()
            }
        }) {
            VStack(spacing: 0) {
                HStack {
                    HStack(spacing: 6) {
                        Text("Total Amount")
                            .font(.system(size: 14, weight: .semibold))
                            .foregroundColor(Color(hex: "#2563EB"))
                        Image(systemName: isTotalExpanded ? "chevron.up" : "chevron.down")
                            .font(.system(size: 12))
                            .foregroundColor(Color(hex: "#2563EB"))
                    }

                    Spacer()

                    Text("₹\(formatPrice(totalAmount))")
                        .font(.system(size: 18, weight: .bold))
                        .foregroundColor(Color(hex: "#2563EB"))
                }

                if isTotalExpanded {
                    VStack(spacing: 6) {
                        Divider()
                            .background(Color(hex: "#BFDBFE"))
                            .padding(.top, 12)

                        HStack {
                            Text("Price")
                                .font(.system(size: 14))
                                .foregroundColor(Color(hex: "#4B5563"))
                            Spacer()
                            Text("₹\(formatPrice(totalAmount + discount))")
                                .font(.system(size: 14))
                                .foregroundColor(Color(hex: "#1F2937"))
                        }
                        .padding(.top, 8)

                        HStack {
                            Text("Discount")
                                .font(.system(size: 14))
                                .foregroundColor(Color(hex: "#4B5563"))
                            Spacer()
                            Text("- ₹\(formatPrice(discount))")
                                .font(.system(size: 14, weight: .semibold))
                                .foregroundColor(Color(hex: "#059669"))
                        }
                    }
                }
            }
            .padding(16)
            .background(Color(hex: "#EFF6FF"))
            .overlay(
                RoundedRectangle(cornerRadius: 8)
                    .stroke(Color(hex: "#DBEAFE"), lineWidth: 1)
            )
            .cornerRadius(8)
        }
        .buttonStyle(PlainButtonStyle())
        .padding(.horizontal, 16)
        .padding(.top, 16)
        .padding(.bottom, 8)
    }

    // MARK: - Discount Banner
    private var discountBanner: some View {
        HStack {
            VStack(alignment: .leading, spacing: 2) {
                Text("Instant Discount")
                    .font(.system(size: 14, weight: .bold))
                    .foregroundColor(Color(hex: "#065F46"))
                Text("Claim now with payment offers")
                    .font(.system(size: 12))
                    .foregroundColor(Color(hex: "#047857"))
            }

            Spacer()

            ZStack {
                Circle()
                    .fill(Color.white)
                    .frame(width: 24, height: 24)
                    .shadow(color: .black.opacity(0.1), radius: 1, y: 1)
                Image(systemName: "tag.fill")
                    .font(.system(size: 10))
                    .foregroundColor(Color(hex: "#047857"))
            }
        }
        .padding(16)
        .background(Color(hex: "#ECFDF5"))
        .cornerRadius(8)
        .padding(.horizontal, 16)
        .padding(.bottom, 16)
    }

    // MARK: - Pay Online Card
    private var payOnlineCard: some View {
        VStack(alignment: .leading, spacing: 16) {
            HStack(spacing: 12) {
                // Shield Icon
                Image(systemName: "shield.checkered")
                    .font(.system(size: 24))
                    .foregroundColor(Color(hex: "#002E6E"))

                VStack(alignment: .leading, spacing: 2) {
                    Text("Pay Online")
                        .font(.system(size: 15, weight: .bold))
                        .foregroundColor(Color(hex: "#1F2937"))
                    Text("UPI, Cards, Wallet, NetBanking")
                        .font(.system(size: 12))
                        .foregroundColor(Color(hex: "#6B7280"))
                }

                Spacer()

                // Recommended Badge
                Text("Recommended")
                    .font(.system(size: 10, weight: .bold))
                    .foregroundColor(.white)
                    .padding(.horizontal, 6)
                    .padding(.vertical, 2)
                    .background(Color(hex: "#BE123C"))
                    .cornerRadius(4)
            }

            Button(action: {
                if !isLoading {
                    onPaymentSelect("RAZORPAY")
                }
            }) {
                if isLoading {
                    HStack(spacing: 8) {
                        ProgressView()
                            .progressViewStyle(CircularProgressViewStyle(tint: .white))
                            .scaleEffect(0.9)
                        Text("Processing...")
                            .font(.system(size: 14, weight: .semibold))
                            .foregroundColor(.white)
                    }
                    .frame(maxWidth: .infinity)
                    .frame(height: 44)
                    .background(Color(hex: "#93C5FD"))
                    .cornerRadius(6)
                } else {
                    Text("Pay Now")
                        .font(.system(size: 14, weight: .semibold))
                        .foregroundColor(.white)
                        .frame(maxWidth: .infinity)
                        .frame(height: 44)
                        .background(Color(hex: "#2563EB"))
                        .cornerRadius(6)
                }
            }
            .disabled(isLoading)
        }
        .padding(16)
        .background(Color.white)
        .cornerRadius(8)
        .shadow(color: .black.opacity(0.05), radius: 2, y: 1)
        .padding(.horizontal, 16)
        .padding(.bottom, 12)
    }

    private func formatPrice(_ price: Double) -> String {
        let formatter = NumberFormatter()
        formatter.numberStyle = .decimal
        formatter.maximumFractionDigits = 0
        return formatter.string(from: NSNumber(value: price)) ?? "\(Int(price))"
    }
}

// MARK: - Payment Accordion View
public struct PaymentAccordionView<Content: View>: View {
    let id: String
    let icon: String
    let title: String
    let subtitle: String?
    let offerText: String?
    @Binding var expandedSection: String?
    @ViewBuilder let content: () -> Content

    init(
        id: String,
        icon: String,
        title: String,
        subtitle: String? = nil,
        offerText: String? = nil,
        expandedSection: Binding<String?>,
        @ViewBuilder content: @escaping () -> Content
    ) {
        self.id = id
        self.icon = icon
        self.title = title
        self.subtitle = subtitle
        self.offerText = offerText
        self._expandedSection = expandedSection
        self.content = content
    }

    var isExpanded: Bool {
        expandedSection == id
    }

    public var body: some View {
        VStack(spacing: 0) {
            // Header
            Button(action: {
                withAnimation(.easeInOut(duration: 0.25)) {
                    expandedSection = isExpanded ? nil : id
                }
            }) {
                HStack(alignment: .top) {
                    HStack(alignment: .top, spacing: 12) {
                        Image(systemName: icon)
                            .font(.system(size: 20))
                            .foregroundColor(Color(hex: "#374151"))
                            .frame(width: 24)

                        VStack(alignment: .leading, spacing: 2) {
                            Text(title)
                                .font(.system(size: 15, weight: .semibold))
                                .foregroundColor(Color(hex: "#111827"))

                            if let subtitle = subtitle {
                                Text(subtitle)
                                    .font(.system(size: 12))
                                    .foregroundColor(Color(hex: "#6B7280"))
                            }

                            if let offerText = offerText {
                                Text(offerText)
                                    .font(.system(size: 12, weight: .medium))
                                    .foregroundColor(Color(hex: "#16A34A"))
                            }
                        }
                    }

                    Spacer()

                    Image(systemName: isExpanded ? "chevron.up" : "chevron.down")
                        .font(.system(size: 14))
                        .foregroundColor(Color(hex: "#6B7280"))
                }
                .padding(16)
                .background(isExpanded ? Color(hex: "#F9FAFB") : Color.white)
            }
            .buttonStyle(PlainButtonStyle())

            // Content
            if isExpanded {
                VStack {
                    content()
                }
                .padding(.horizontal, 16)
                .padding(.leading, 36)
                .padding(.bottom, 16)
                .padding(.top, 4)
                .background(Color.white)
            }
        }
        .background(Color.white)
    }
}

// MARK: - Preview
struct PaymentView_Previews: PreviewProvider {
    static var previews: some View {
        PaymentView(
            totalAmount: 1499,
            discount: 200,
            itemCount: 1,
            onBack: {},
            onPaymentSelect: { _ in },
            isLoading: .constant(false)
        )
    }
}

File: apps/Applications/android/app/src/main/java/com/ecommerceearn/app/ui/pages/PaymentView.kt
package com.ecommerceearn.app.ui.pages

import androidx.compose.animation.*
import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.verticalScroll
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.*
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.shadow
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import java.text.NumberFormat
import java.util.*

@Composable
fun PaymentView(
    totalAmount: Double,
    discount: Double,
    _itemCount: Int,
    onBack: () -> Unit,
    onPaymentSelect: (String) -> Unit,
    isLoading: Boolean
) {
    var isTotalExpanded by remember { mutableStateOf(false) }
    var expandedSection by remember { mutableStateOf<String?>(null) }

    Column(
        modifier = Modifier
            .fillMaxSize()
            .background(Color(0xFFF3F4F6))
    ) {
        // Header
        Row(
            modifier = Modifier
                .fillMaxWidth()
                .background(Color.White)
                .padding(horizontal = 16.dp, vertical = 12.dp),
            verticalAlignment = Alignment.CenterVertically
        ) {
            IconButton(onClick = onBack, modifier = Modifier.size(32.dp)) {
                Icon(Icons.Default.ArrowBack, contentDescription = "Back", tint = Color(0xFF1F2937))
            }
            Spacer(modifier = Modifier.width(12.dp))
            Column {
                Text("Step 3 of 3", fontSize = 12.sp, color = Color(0xFF6B7280))
                Text("Payments", fontSize = 18.sp, fontWeight = FontWeight.Bold, color = Color(0xFF1F2937))
            }
            Spacer(modifier = Modifier.weight(1f))
            Row(
                modifier = Modifier
                    .background(Color(0xFFF3F4F6), RoundedCornerShape(4.dp))
                    .padding(horizontal = 8.dp, vertical = 4.dp),
                verticalAlignment = Alignment.CenterVertically
            ) {
                Icon(Icons.Default.Lock, contentDescription = "Secure", tint = Color(0xFF4B5563), modifier = Modifier.size(10.dp))
                Spacer(modifier = Modifier.width(4.dp))
                Text("100% Secure", fontSize = 12.sp, fontWeight = FontWeight.SemiBold, color = Color(0xFF4B5563))
            }
        }
        HorizontalDivider(color = Color(0xFFE5E7EB))

        Column(
            modifier = Modifier
                .fillMaxSize()
                .verticalScroll(rememberScrollState())
        ) {
            // Total Amount Card
            Column(
                modifier = Modifier
                    .padding(horizontal = 16.dp, vertical = 16.dp)
                    .background(Color(0xFFEFF6FF), RoundedCornerShape(8.dp))
                    .border(1.dp, Color(0xFFDBEAFE), RoundedCornerShape(8.dp))
                    .clickable { isTotalExpanded = !isTotalExpanded }
                    .padding(16.dp)
            ) {
                Row(verticalAlignment = Alignment.CenterVertically) {
                    Text("Total Amount", fontSize = 14.sp, fontWeight = FontWeight.SemiBold, color = Color(0xFF2563EB))
                    Icon(
                        imageVector = if (isTotalExpanded) Icons.Default.KeyboardArrowUp else Icons.Default.KeyboardArrowDown,
                        contentDescription = null,
                        tint = Color(0xFF2563EB),
                        modifier = Modifier.size(16.dp)
                    )
                    Spacer(modifier = Modifier.weight(1f))
                    Text("₹${formatPrice(totalAmount)}", fontSize = 18.sp, fontWeight = FontWeight.Bold, color = Color(0xFF2563EB))
                }

                AnimatedVisibility(visible = isTotalExpanded, enter = expandVertically(), exit = shrinkVertically()) {
                    Column(modifier = Modifier.padding(top = 12.dp)) {
                        HorizontalDivider(color = Color(0xFFBFDBFE))
                        Row(modifier = Modifier.padding(top = 8.dp)) {
                            Text("Price", fontSize = 14.sp, color = Color(0xFF4B5563))
                            Spacer(modifier = Modifier.weight(1f))
                            Text("₹${formatPrice(totalAmount + discount)}", fontSize = 14.sp, color = Color(0xFF1F2937))
                        }
                        Row(modifier = Modifier.padding(top = 4.dp)) {
                            Text("Discount", fontSize = 14.sp, color = Color(0xFF4B5563))
                            Spacer(modifier = Modifier.weight(1f))
                            Text("- ₹${formatPrice(discount)}", fontSize = 14.sp, fontWeight = FontWeight.SemiBold, color = Color(0xFF059669))
                        }
                    }
                }
            }

            if (discount > 0) {
                Row(
                    modifier = Modifier
                        .padding(horizontal = 16.dp)
                        .padding(bottom = 16.dp)
                        .background(Color(0xFFECFDF5), RoundedCornerShape(8.dp))
                        .padding(16.dp),
                    verticalAlignment = Alignment.CenterVertically
                ) {
                    Column {
                        Text("Instant Discount", fontSize = 14.sp, fontWeight = FontWeight.Bold, color = Color(0xFF065F46))
                        Text("Claim now with payment offers", fontSize = 12.sp, color = Color(0xFF047857))
                    }
                    Spacer(modifier = Modifier.weight(1f))
                    Box(modifier = Modifier.size(24.dp).background(Color.White, CircleShape).shadow(1.dp, CircleShape), contentAlignment = Alignment.Center) {
                        Icon(Icons.Default.Sell, contentDescription = null, tint = Color(0xFF047857), modifier = Modifier.size(12.dp))
                    }
                }
            }

            Text(
                "Payment Options",
                fontSize = 14.sp,
                fontWeight = FontWeight.SemiBold,
                color = Color(0xFF374151),
                modifier = Modifier.padding(horizontal = 16.dp).padding(bottom = 12.dp)
            )

            // Pay Online Card
            Column(
                modifier = Modifier
                    .padding(horizontal = 16.dp).padding(bottom = 12.dp)
                    .background(Color.White, RoundedCornerShape(8.dp))
                    .shadow(2.dp, RoundedCornerShape(8.dp))
                    .padding(16.dp)
            ) {
                Row(verticalAlignment = Alignment.CenterVertically, modifier = Modifier.padding(bottom = 16.dp)) {
                    Icon(Icons.Default.Security, contentDescription = null, tint = Color(0xFF002E6E), modifier = Modifier.size(24.dp))
                    Spacer(modifier = Modifier.width(12.dp))
                    Column {
                        Text("Pay Online", fontSize = 15.sp, fontWeight = FontWeight.Bold, color = Color(0xFF1F2937))
                        Text("UPI, Cards, Wallet, NetBanking", fontSize = 12.sp, color = Color(0xFF6B7280))
                    }
                    Spacer(modifier = Modifier.weight(1f))
                    Text(
                        "Recommended",
                        fontSize = 10.sp,
                        fontWeight = FontWeight.Bold,
                        color = Color.White,
                        modifier = Modifier.background(Color(0xFFBE123C), RoundedCornerShape(4.dp)).padding(horizontal = 6.dp, vertical = 2.dp)
                    )
                }

                Button(
                    onClick = { if (!isLoading) onPaymentSelect("RAZORPAY") },
                    enabled = !isLoading,
                    modifier = Modifier.fillMaxWidth().height(44.dp),
                    shape = RoundedCornerShape(6.dp),
                    colors = ButtonDefaults.buttonColors(containerColor = if (isLoading) Color(0xFF93C5FD) else Color(0xFF2563EB))
                ) {
                    if (isLoading) {
                        CircularProgressIndicator(color = Color.White, modifier = Modifier.size(20.dp), strokeWidth = 2.dp)
                        Spacer(modifier = Modifier.width(8.dp))
                        Text("Processing...")
                    } else {
                        Text("Pay Now")
                    }
                }
            }

            // Saved Payment Options
            PaymentAccordionView(
                id = "saved",
                icon = Icons.Default.Schedule,
                title = "Saved Payment Options",
                expandedSection = expandedSection,
                onToggle = { expandedSection = if (expandedSection == "saved") null else "saved" }
            ) {
                Text("No saved cards found.", fontSize = 13.sp, color = Color(0xFF9CA3AF))
            }

            // Cash on Delivery
            PaymentAccordionView(
                id = "cod",
                icon = Icons.Default.LocalAtm,
                title = "Cash on Delivery",
                expandedSection = expandedSection,
                onToggle = { expandedSection = if (expandedSection == "cod") null else "cod" }
            ) {
                Button(
                    onClick = { onPaymentSelect("COD") },
                    modifier = Modifier.fillMaxWidth().height(44.dp),
                    shape = RoundedCornerShape(6.dp),
                    colors = ButtonDefaults.buttonColors(containerColor = Color(0xFF2563EB))
                ) {
                    Text("Place Order", fontWeight = FontWeight.Bold, color = Color.White)
                }
            }
            
            Spacer(modifier = Modifier.height(40.dp))
        }
    }
}

@Composable
fun PaymentAccordionView(
    id: String,
    icon: androidx.compose.ui.graphics.vector.ImageVector,
    title: String,
    subtitle: String? = null,
    offerText: String? = null,
    expandedSection: String?,
    onToggle: () -> Unit,
    content: @Composable () -> Unit
) {
    val isExpanded = expandedSection == id
    Column(modifier = Modifier.background(Color.White)) {
        Row(
            modifier = Modifier
                .fillMaxWidth()
                .background(if (isExpanded) Color(0xFFF9FAFB) else Color.White)
                .clickable { onToggle() }
                .padding(16.dp),
            verticalAlignment = Alignment.Top
        ) {
            Icon(icon, contentDescription = null, tint = Color(0xFF374151), modifier = Modifier.size(20.dp))
            Spacer(modifier = Modifier.width(12.dp))
            Column {
                Text(title, fontSize = 15.sp, fontWeight = FontWeight.SemiBold, color = Color(0xFF111827))
                subtitle?.let { Text(it, fontSize = 12.sp, color = Color(0xFF6B7280)) }
                offerText?.let { Text(it, fontSize = 12.sp, fontWeight = FontWeight.Medium, color = Color(0xFF16A34A)) }
            }
            Spacer(modifier = Modifier.weight(1f))
            Icon(
                imageVector = if (isExpanded) Icons.Default.KeyboardArrowUp else Icons.Default.KeyboardArrowDown,
                contentDescription = null,
                tint = Color(0xFF6B7280),
                modifier = Modifier.size(14.dp)
            )
        }

        AnimatedVisibility(visible = isExpanded, enter = expandVertically(), exit = shrinkVertically()) {
            Box(modifier = Modifier.background(Color.White).padding(start = 48.dp, end = 16.dp, bottom = 16.dp, top = 4.dp)) {
                content()
            }
        }
    }
}

private fun formatPrice(price: Double): String {
    val format = NumberFormat.getNumberInstance(Locale("en", "IN"))
    format.maximumFractionDigits = 0
    return format.format(price)
}
